.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,67 @@
+name: zzz [DO NOT RUN] Automated unit tests
+
+on:
+  workflow_dispatch: # TODO: remove this after testing
+  pull_request:
+    branches:
+      - dev
+      - main
+  push:
+    branches:
+      - dev
+      - main
+
```

Comment:
I would suggest to add a couple `path-ignore` patters here, e.g.
```yaml
paths-ignore:
      - '**.md'
      - '**/README'
      - '**.yml'
      - '**.txt'
```
for which we do not need to run swift-based unit tests.